src/app-layer-modbus.c

```diff
@@ -395,6 +394,21 @@ void ModbusStateTxFree(void *state, uint64_t tx_id) {
     SCReturn;
 }
 
+const char * ModbusGetFunctionName(uint8_t function)
+{
+    return SCMapEnumValueToName(function, modbus_function_name_table);
```

Comment:
I'm ok with this for modbus, but you have to realize this isn't very fast. It essentially does a linear search.